agent.py

```diff
@@ -78,9 +83,32 @@ def _handle_sigterm(self, signum, frame):
         log.debug("Collector is stopped.")
 
     def _handle_sigusr1(self, signum, frame):
+        """Handles SIGUSR1, which signals an exit with an autorestart."""
         self._handle_sigterm(signum, frame)
         self._do_restart()
 
+    def _handle_sigusr2(self, signum, frame):
+        """Handles SIGUSR2, which signals a configuration reload."""
+        log.info("Attempting a configuration reload...")
+        log.info("AGENT.PY SIGUSR2: self.collector: " + str(self.collector))
+        self.reload_configs()
+
+    def reload_configs(self):
+        """Reloads the agent configuration and checksd configurations."""
+        # Reload checksd configs
+        hostname = get_hostname(self._agentConfig)
+        self._checksd = load_check_directory(self._agentConfig, hostname)
+
+        # Logging
+        num_checks = len(self._checksd['initialized_checks'])
+        if num_checks > 0:
+            log.info("SIGUSR2: Successfully reloaded {num_checks} checks".
+                     format(num_checks=num_checks))
+            for che in self._checksd['initialized_checks']:
```

Comment:
nitpick: when there is no real need to shorten the variable names, there is no need :) always prefer clarity to brevity if the performance is the same